libs/scan.py

```diff
@@ -70,159 +73,118 @@ def draw_sg(security_group, sg_raw):
-        'http://acs.amazonaws.com/groups/s3/LogDelivery': 'ACL: S3 Log Delivery'
-    }
-    for grant in acls:
-        if isinstance(permission, list) and grant['Permission'] not in permission:
-            continue
-        if isinstance(permission, str) and grant['Permission'] != permission:
-            continue
-        if 'URI' in grant['Grantee'] and grant['Grantee']['URI'] in map_users_uri:
-            if override or map_users_uri[grant['Grantee']['URI']] not in s3_report:
-                s3_report[map_users_uri[grant['Grantee']['URI']]] = right
-            else:
-                s3_report[map_users_uri[grant['Grantee']['URI']]] = f'{s3_report[map_users_uri[grant["Grantee"]["URI"]]]},{right}'
-            if map_users_uri[grant['Grantee']['URI']] in ['ACL: All Users', 'ACL: Any Authenticated Users']:
-                s3_report['PubliclyAccessible'] = True
-    return s3_report
-
-def s3_scan_acls(s3_report, acls):
-    """
-    Scan ACLS for S3 Buckets
-    """
-    s3_report = s3_scan_concat_permissions(s3_report, acls, 'READ', 'LIST')
-    s3_report = s3_scan_concat_permissions(s3_report, acls, 'READ_ACP', 'READ')
-    s3_report = s3_scan_concat_permissions(s3_report, acls, ['WRITE', 'WRITE_ACP'], 'WRITE')
-    s3_report = s3_scan_concat_permissions(
-        s3_report,
-        acls,
-        'FULL_CONTROL',
-        'LIST,READ,WRITE',
-        override=True)
-    return s3_report
+    for i, asset in enumerate(assets):
+        asset_type = asset.get_type()
+        if asset_type == 'EC2' and record_value in (asset.public_ip, asset.private_ip):
+            assets[i].dns_record = record['Name'].replace('\\052', '*')
+        elif asset_type == 'ELBV2' and record_value == f'{asset.name}.':
+            assets[i].dns_record = record['Name'].replace('\\052', '*')
+    return assets
 
-def s3_scan(report, s_three, configuration, location, acls, public_only):
+def s3_scan(s_three, configuration, region, acls, public_only):
     """
     Scan S3 Buckets
     """
-    is_private = False
-    if configuration is not None:
-        is_private = configuration['BlockPublicAcls'] and \
-            configuration['IgnorePublicAcls'] and \
-            configuration['BlockPublicPolicy'] and \
-            configuration['RestrictPublicBuckets']
-    if public_only and is_private:
-        return report
-    if location not in report:
-        report[location] = dict()
-        report[location]['S3'] = dict()
-    report[location]['S3'][s_three] = dict()
-    report[location]['S3'][s_three]['Type'] = 'S3'
-    report[location]['S3'][s_three]['Name'] = f's3://{s_three}'
-    report[location]['S3'][s_three]['URL'] = f'https://{s_three}.s3.{location}.amazonaws.com/'
-    # Can be erased in 's3_scan_concat_permissions' function
-    report[location]['S3'][s_three]['PubliclyAccessible'] = False
+    s3_asset = S3(
+        name=f's3://{s_three}',
+        url=f'https://{s_three}.s3.{region}.amazonaws.com/')
```

Comment:
Could be good to add these URI in a config file I think?

Reply:
Why ?

Reply:
It may be easier to keep a config file only for this.
For example if you define in multiple location this URI and AWS change it, it will be harder to maintain it

But if you are sure to use it only here, then it's not a big deal :)

Reply:
Yes, I understand but for the moment I'll keep it here :/
Thanks for the feedback though